base/base32768_test.cpp

```diff
@@ -0,0 +1,246 @@
+﻿
+#include "base/base32768.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/array.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace principia {
+namespace base {
+
+class Base32768Test : public testing::Test {
+ protected:
+  void CheckEncoding(char const* const binary,
+                     int const binary_length,
+                     const char16_t* const base32768) {
+    int const base32768_length = std::char_traits<char16_t>::length(base32768);
+    Array<std::uint8_t const> input(
+        reinterpret_cast<std::uint8_t const*>(binary), binary_length);
+    UniqueBytes output(sizeof(char16_t) * base32768_length);
+    Base32768Encode(input, output.get());
+    char16_t const* const output32768 =
+        reinterpret_cast<char16_t*>(output.get().data);
+    EXPECT_EQ(0,
+              std::char_traits<char16_t>::compare(
+                  output32768, base32768, base32768_length));
+    for (int i = 0; i < base32768_length; ++i) {
+      EXPECT_EQ(output32768[i], base32768[i])
+          << "index=" << i << " actual=" << std::hex << output32768[i]
+          << " expected=" << base32768[i];
+    }
+  }
+};
+
+using Base32768DeathTest = Base32768Test;
+
+TEST_F(Base32768Test, EncodeCaseDemo) {
```

Comment:
`EncodeEmptyStringMD5`. Append a 7E to `binary`, and add an additional `CheckEncoding` that encodes all 16 bytes (which should result in "遮視塀⤠䶌Ԇ堹麢" followed by an additional character, since 15 * 8 ≡ 0 (mod 15).